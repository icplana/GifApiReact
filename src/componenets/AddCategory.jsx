// import React from 'react' no necesario desde React 17
import { useState } from "react"
import { PropTypes } from 'prop-types' //para poder usar hay que hacer yarn add prop-types o el respectivo npm
export const AddCategory = ({ onNewCategory }) => {
    
    const [ inputValue, setInputValue ] = useState('')
    
    // const onInputChange = (event) => {
    //     setInputValue(event.target.value)
    // }
    const onInputChange = ({ target }) => {
        setInputValue( target.value )
    }

    const onSubmit = event => {
        event.preventDefault()
        if( inputValue.trim().length <= 1) return
        // setCategories( categories => [...categories, inputValue])
        onNewCategory( inputValue.trim())
        setInputValue('')
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text" 
                placeholder="Buscar gifs"
                value = { inputValue }
                onChange={ onInputChange }
            />
            <button
            onClick={ onSubmit }>Agregar</button>
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
