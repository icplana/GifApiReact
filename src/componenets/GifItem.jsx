import PropTypes from 'prop-types'

export const GifItem = ({title, url}) => {
  return (
    <div className="card" >
        <img  src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}


/*
Tarea:

1. AÃ±adir proptypes:
  - title obligatorio
  - url obligatoria

2. Evaluar snapshot
*/


GifItem.propTypes = {
  title : PropTypes.string.isRequired,
  url : PropTypes.string.isRequired
}