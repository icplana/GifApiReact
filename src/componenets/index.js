//archivo de barrill

export * from './AddCategory'
export * from './GifGrid'
export * from './GifItem'