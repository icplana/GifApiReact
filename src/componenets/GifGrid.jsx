import { GifItem } from "./GifItem"
import PropTypes from 'prop-types'
import { useFetchGifs } from "../hooks/useFetchGifs"
import { GifExpertApp } from "../GifExpertApp"
export const GifGrid = ({ category, onDelCategory }) => {

    //Hook personalizado
    const { images, isLoading } = useFetchGifs( category )

    // const deleteCategory = (e) => {
    //     // const cat = e.target.previousSibling.innerHTML
    //     // const temporalArr = arr.filter( category => category !== cat )
    //     // setCategories(temporalArr)
    //     e.target.nextSibling.remove()
    //     e.target.previousSibling.remove()
    //     e.target.remove()
        
    // }

    return (
        <>
            <h3>{ category }</h3>
            
            {/* <button onClick={ GifExpertApp(deleteCategory) }>Delete</button> */}
               
                    <svg onClick={ onDelCategory } xmlns="http://www.w3.org/2000/svg"  width="20" height="20" viewBox="0 0 24 24">
                        <path d="M7 4V2H17V4H22V6H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V6H2V4H7ZM6 6V20H18V6H6ZM9 9H11V17H9V9ZM13 9H15V17H13V9Z">
                        </path>
                    </svg>
          
            {
                isLoading && <h2>Cargando...</h2>
            }
            <div className="card-grid">
                {
                    images.map( img => (
                        <GifItem 
                            key= {img.id} 
                            { ...img }
                        /> 
                    ))
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    // onDelCategory: PropTypes.func.isRequired
}