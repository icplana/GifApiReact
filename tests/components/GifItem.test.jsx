import { screen, render } from "@testing-library/react"
import { GifItem } from "../../src/componenets/GifItem"
describe('testing GifItem component', () => {
    const title = 'El titulo'
    const url = 'https://unaurl.com/'
    test('testing match snapshot', () => {
        
       
        const { container } = render( <GifItem title = { title } url={ url } />)
        expect( container ).toMatchSnapshot()
        
    
    })


    test('debe mostrar la imagen con el url y el alt indicado', () => {
        render( <GifItem url = {url} title={title}/>)

        // screen.debug()

        expect(screen.getByRole('img').src).toEqual(url)
        expect(screen.getByRole('img').alt).toEqual(title)
        //para simplificar codigo cuando hacemos varios "expect" (analizamos varios atributos o de varias formas) podemos hacerlo de la siguiente manera (desestructurando el objeto):
        const {src, alt} = screen.getByRole('img')
        expect( src ).toBe( src )
        expect( alt ).toBe( alt )
    })

    test('chequear el texto del parrafo', () => {
        render( <GifItem title = { title } url = { url }/>)
        expect( screen.getByText( title ) ).toBeTruthy() //Aqui deberia ser m√°s especifico en muchos casos pero en este tenemos suficiente con esto
    })
})