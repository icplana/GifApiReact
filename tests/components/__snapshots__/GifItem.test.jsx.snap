// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifItem component testing match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="El titulo"
      src="https://unaurl.com/"
    />
    <p>
      El titulo
    </p>
  </div>
</div>
`;
